services/prometheus/prometheus_test.go

```diff
@@ -518,3 +521,39 @@ scrape_configs:
 		assert.Equal(t, expected, string(actual), "actual:\n%s", actual)
 	})
 }
+
+func TestBasePrometheusConfig(t *testing.T) {
+	t.Run("Default", func(t *testing.T) {
+		db, svc, original := setup(t)
+		defer teardown(t, db, svc, original)
+
+		assert.NoError(t, svc.updateConfiguration())
+		tmpfile, err := ioutil.TempFile("", "mock_base_config.yml")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		defer os.Remove(tmpfile.Name())
+		oldBasePrometheusConfig := basePrometheusConfig
+		basePrometheusConfig = tmpfile.Name()
+		cfg := []byte(`# Managed by pmm-managed. DO NOT EDIT.
+---
+global:
+  scrape_interval: 9m
+  scrape_timeout: 19s
+  evaluation_interval: 9m
+rule_files:
+- /srv/prometheus/rules/fake_first.rules.yml`)
+
+		assert.NoError(t, ioutil.WriteFile(basePrometheusConfig, cfg, os.ModePerm))
+
+		newcfg, err := svc.marshalConfig()
+		assert.NoError(t, err)
+
+		assert.True(t, bytes.Contains(newcfg, cfg))
```

Comment:
@AlekSi is it ok that base config file overrides our global configs?

Reply:
Yes, because our GlobalConfig settings are always present; base file is the only way to change them.